### PR TITLE
Refactoring adder_atlas_plugin.py: modernizing code where possible

### PR DESCRIPTION
**Pull request by EdwardKaravakis**

### CODE REVIEW

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 1]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+                                    tmp_dest = tmp_se_tokens[tmp_dest_token]
+                                else:
+                                    tmp_dest = src_site_spec.ddm_output[scope_src_site_spec_output]
+                                if tmp_dest not in tmp_dest_list:
+                                    tmp_dest_list.append(tmp_dest)
+                        # add
+                        dataset_destination_map[file_destination_dispatch_block] = tmp_dest_list
+
+                # extra meta data
+                if self.ddm_backend == "rucio":
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:379]: AFAIU the ddm_backend was a transition variable from LFC/DQ2/Rucio, but right now there is only rucio (sometimes the backend is not set explicitly, but it's defaulted to Rucio). Do we need these if's?

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 2]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+            if not is_zip_file and not self.add_to_top_only:
+                for tmp_zip_file_name, tmp_zip_contents in zip_file_map.items():
+                    if any(re.search(f"^{tmp_zip_content}$", file.lfn) for tmp_zip_content in tmp_zip_contents):
+                        zip_file_name = tmp_zip_file_name
+                        break
+                if zip_file_name:
+                    zip_files.setdefault(zip_file_name, {})
+                    cont_zip_map[file.lfn] = zip_file_name
+            try:
+                # check nevents
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:286]: Candidate to move to separate function for nevents, filesize, etc retrieval

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 3]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+                    id_map[file_destination_dispatch_block] = []
+                file_attrs = {
+                    "guid": file.GUID,
+                    "lfn": file.lfn,
+                    "size": fsize,
+                    "checksum": file.checksum,
+                    "ds": file_destination_dispatch_block,
+                }
+                # add SURLs if LFC registration is required
+                if self.use_central_lfc():
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:330]: I find the use_central_lfc function confusing (i.e. LFC was something very old, not sure if Rucio is considered a central LFC and not sure this is related at all to LFCs at the moment). The function just checks the add_to_top_only variable - maybe we can do if directly against add_to_top_only and remove this function?

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 4]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+                        file_attrs["events"] = self.extra_info["nevents"][file.lfn]
+                    elif self.extra_info["nevents"] != {}:
+                        file_attrs["events"] = None
+                    # if file.jediTaskID not in [0,None,'NULL']:
+                    #    fileAttrs['task_id'] = file.jediTaskID
+                    file_attrs["panda_id"] = file.PandaID
+                    if campaign not in ["", None]:
+                        file_attrs["campaign"] = campaign
+
+                # extract OS files
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:392]: What's an OS file? OS=Object Store? 10 years ago I think there was some Event Service development with Object Stores, but I think the experience with OS was not satisfactory and was broken off. Is this related? Is it still in use?

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 5]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+                                            ddm_id = tmp_src_ddm
+                                            # use the first token's location as source for T1D1
+                                            tmp_src_token = file.destinationDBlockToken.split(",")[0]
+                                            if tmp_src_token in src_site_spec.setokens_output[scope_src_site_spec_output]:
+                                                ddm_id = src_site_spec.setokens_output[scope_src_site_spec_output][tmp_src_token]
+                                            opt_source[ddm_id] = {"policy": 0}
+                                        # T1 used as T2
+                                        if (
+                                            src_site_spec.cloud != self.job.cloud
+                                            and (not tmp_src_ddm.endswith("PRODDISK"))
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:485]: Did you check in CRIC if there are RSEs that end in PRODDISK or of ATLASPRODDISK type? I think we had been removing PRODDISK code in the previous exercises.

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 6]
```diff
@@ -691,18 +686,21 @@ def update_outputs(self):
...
+
         # release some memory
         del dest_id_map
         del dataset_destination_map
         del map_for_alt_stage_out
         del zip_files
         del cont_zip_map
         gc.collect()
+
         # register dataset subscription
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:698]: Candidate for new function that registers dataset subscriptions?

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 7]
```diff
@@ -606,21 +603,19 @@ def update_outputs(self):
...
+        reg_file_list = {tmp_reg_item["lfn"] for tmp_reg_list in id_map.values() for tmp_reg_item in tmp_reg_list}
+        reg_num_files = len(reg_file_list)
+
         # decompose idMap
         if not self.use_central_lfc():
             dest_id_map = {None: id_map}
         else:
             dest_id_map = self.decompose_id_map(id_map, dataset_destination_map, map_for_alt_stage_out, sub_to_ds_map, alt_staged_files)
+
         # add files
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:619]: Candidate for new function to add files?

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 8]
```diff
@@ -224,6 +221,7 @@ def update_outputs(self):
                         if re.search("^" + tmp_zip_content + "$", tmp_lfn) is not None and tmp_lfn in n_events_input and n_events_input[tmp_lfn] is not None:
                             self.extra_info["nevents"].setdefault(tmp_zip_file_name, 0)
                             self.extra_info["nevents"][tmp_zip_file_name] += n_events_input[tmp_lfn]
+
         # check files
         id_map = {}
         # fileList = []
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:227]: Remove commented out code

File: pandaserver/dataservice/adder_atlas_plugin.py [hunk 9]
```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
...
+                        zip_files[file.lfn]["scope"] = file.scope
+                        zip_files[file.lfn]["rse"] = dataset_destination_map[file_destination_dispatch_block]
+
+                # for subscription
                 if (
-                    self.job.jobStatus == "finished"
-                    and EventServiceUtils.isEventServiceJob(self.job)
-                    and not EventServiceUtils.isJobCloningJob(self.job)
-                    and file.type != "log"
+                    self.job.prodSourceLabel in ["managed", "test", "software", "user", "rucio_test"] + JobUtils.list_ptest_prod_sources
```
Comment (fbarreir) [inline pandaserver/dataservice/adder_atlas_plugin.py:423]: I can't think of any other prodsourcelabel, not clear what's the point of this line. Was it to exclude the 'ddm' prodsourcelabel in the past?

('software' doesn't exist anymore)

File: pandaserver/dataservice/adder_atlas_plugin.py (hunk 5 above)
Comment (fbarreir) [reply pandaserver/dataservice/adder_atlas_plugin.py:485]: There are a few more PRODDISK things around the file

File: pandaserver/dataservice/adder_atlas_plugin.py (hunk 1 above)
Comment (fbarreir) [reply pandaserver/dataservice/adder_atlas_plugin.py:379]: [root@aipanda098 panda]# grep "ddm backend " panda-adder.log | grep -v rucio
<nothing>

Comment (fbarreir) [reply pandaserver/dataservice/adder_atlas_plugin.py:379]: So there is also potential to remove the stuff around line 70

**Review by fbarreir — COMMENTED** [review]
@tmaeno please check some of my comments where I'm not sure if some blocks should be obsoleted

File: pandaserver/dataservice/adder_atlas_plugin.py (hunk 4 above)
Comment (tmaeno) [reply pandaserver/dataservice/adder_atlas_plugin.py:392]: This was originally implemented for files uploaded to object stores, but this code block is still used for files uploaded to alternative RSEs.

File: pandaserver/dataservice/adder_atlas_plugin.py (hunk 9 above)
Comment (tmaeno) [reply pandaserver/dataservice/adder_atlas_plugin.py:423]: The line excludes jobs with prodsourcelabel=panda, but the next line essentially excludes them. I think OK to remove the condition with prodsourcelabel.
